cli/src/common/boto3/cfn.py

```diff
@@ -105,4 +105,14 @@ def list_pcluster_stacks(self):
     @AWSExceptionHandler.handle_client_exception
     def describe_stack_resource(self, stack_name: str, logic_resource_id: str):
         """Get stack resource information."""
-        self._client.describe_stack_resource(StackName=stack_name, LogicalResourceId=logic_resource_id)
+        return self._client.describe_stack_resource(StackName=stack_name, LogicalResourceId=logic_resource_id)
+
+    @AWSExceptionHandler.handle_client_exception
+    def list_imagebuilder_stacks(self):
+        """List existing imagebuilder stacks."""
+        stack_list = []
+        for stack in self._paginate_results(self._client.describe_stacks):
+            tag = next(iter([tag["Value"] for tag in stack.get("Tags") if tag["Key"] == PCLUSTER_IMAGE_NAME_TAG]), None)
+            if stack.get("ParentId") is None and tag:
```

Comment:
You can optimize the call by filtering out stacks on which ParentId is None before iterating tags.
```python
if stack.get("ParentId") is None and next(iter([tag["Value"] for tag in stack.get("Tags") if tag["Key"] == PCLUSTER_IMAGE_NAME_TAG]), None):
    stack_list.append(...)
```

or 
```python
if stack.get("ParentId") is None:
    if next(iter([tag["Value"] for tag in stack.get("Tags") if tag["Key"] == PCLUSTER_IMAGE_NAME_TAG]), None):
      stack_list.append(...)
```

Reply:
You have other alternatives to improve this code:
* Use list comprehension, see `list_pcluster_stacks`
* Return `StackInfo` objects and use `stack_info._get_tag(PCLUSTER_IMAGE_NAME_TAG)` rather than rewriting the get tag function

Reply:
done